routing/src/vrf.rs

```diff
@@ -0,0 +1,447 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+//! VRF module to store Ipv4 and Ipv6 routing tables
+
+use std::hash::Hash;
+use std::net::IpAddr;
+#[cfg(test)]
+use std::str::FromStr;
+
+use crate::nexthop::{Nhop, NhopKey, NhopStore, Rc};
+use crate::prefix::Prefix;
+use iptrie::{Ipv4Prefix, Ipv6Prefix, RTrieMap};
+use net::vxlan::Vni;
+
+/// We'll use the RPC definitions for these
+use dplane_rpc::msg::{RouteDistance, RouteMetric, RouteType};
+
+/// Every VRF is univocally identified with a numerical VRF id
+pub type VrfId = u32;
+
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+#[doc = "A next-hop in the VRF"]
+pub struct RouteNhop {
```

Comment:
you can just use `///`, no need for `#[doc = ]` unless you want to do something more fancy in terms of preprocessing the docs.

Reply:
Agreed.